packages/bus/src/Brokers/RMQEventBus.ts

```diff
@@ -0,0 +1,395 @@
+    const heartbeat = this.configuration.heartbeatInterval;
+    const credentials = this.configuration.credentials;
+    const bus = this.configuration.bus;
+
+    const buildedUrl = this.buildUrl(bus, credentials, heartbeat || 5000);
+
+    this.client = await this.backOffRetry.acquire(async () =>
+      this.connectionFactory.create(buildedUrl, {
+        timeout: connectTimeout
+      })
+    );
+
+    await this.createConsumerChannel();
+  }
+
+  public async register<T, R>(
+    type: EventHandlerConstructor<T, R>
+  ): Promise<void> {
+    if (!this.channel) {
+      throw new IllegalOperation(this);
+    }
+
+    if (!this.container.isRegistered(type)) {
+      throw new EventHandlerNotFound(type.name);
+    }
+    const handlerInstance = this.container.resolve(type);
+
+    const eventNames: string[] = this.reflectEventsNames(type);
+    if (eventNames?.length === 0) {
+      throw new NoSubscriptionsFound(handlerInstance);
+    }
+
+    for (const eventName of eventNames) {
+      const eventHandlers = this.handlers.get(eventName);
+      if (Array.isArray(eventHandlers)) {
+        eventHandlers.push(handlerInstance);
+      } else {
+        await this.channel?.bindQueue(
+          this.configuration.clientQueue,
+          this.configuration.exchange,
+          eventName
+        );
+        this.handlers.set(eventName, [handlerInstance]);
+      }
+    }
+  }
+
+  public async unregister<T, R>(
+    type: EventHandlerConstructor<T, R>
+  ): Promise<void> {
+    if (!this.channel) {
+      throw new IllegalOperation(this);
+    }
+
+    if (!this.container.isRegistered(type)) {
+      throw new EventHandlerNotFound(type.name);
+    }
+    const handlerInstance = this.container.resolve(type);
+
+    const eventNames: string[] = this.reflectEventsNames(type);
+    if (eventNames?.length === 0) {
+      throw new NoSubscriptionsFound(handlerInstance);
+    }
+
+    for (const eventName of eventNames) {
+      const eventHandlers = this.handlers.get(eventName);
+      if (!eventHandlers) {
+        continue;
+      }
+
+      const handlerIndex = eventHandlers.indexOf(handlerInstance);
+      eventHandlers.splice(handlerIndex, 1);
+
+      await this.channel?.unbindQueue(
+        this.configuration.clientQueue,
+        this.configuration.exchange,
+        eventName
+      );
+    }
+  }
+
+  public async publish<T>(event: Event<T>): Promise<void> {
+    const { type, payload, correlationId, createdAt } = event;
+    if (!this.channel) {
+      throw new IllegalOperation(this);
+    }
+
+    await this.backOffRetry.acquire(() => {
+      this.channel?.publish(
+        this.configuration.exchange,
+        type,
+        Buffer.from(JSON.stringify(payload)),
+        {
+          contentType: 'application/json',
+          mandatory: true,
+          persistent: true,
+          replyTo: this.REPLY_QUEUE_NAME,
+          correlationId,
+          timestamp: createdAt.getTime()
+        }
```

Comment:
The shorthand property should be placed first

```suggestion
        {
          correlationId,
          contentType: 'application/json',
          mandatory: true,
          persistent: true,
          replyTo: this.REPLY_QUEUE_NAME,
          timestamp: createdAt.getTime()
        }
```